pkgs/top-level/python-packages.nix

```diff
@@ -9336,6 +9336,11 @@ in {
 
     buildInputs = with self; [ nose sphinx numpydoc ];
 
+    # Failing test on Darwin
+    patchPhase = '''' + optionalString stdenv.isDarwin ''
+      sed -i 's/test_dispatch_multiprocessing/_test_dispatch_multiprocessing/' joblib/test/test_parallel.py
+    '';
+
```

Comment:
Like this?

    postPatch = '''' + optionalString isPy3k ''
      sed -i -e '70,84d' joblib/test/test_format_stack.py
    '' + optionalString stdenv.isDarwin ''
      sed -i 's/test_dispatch_multiprocessing/_test_dispatch_multiprocessing/' joblib/test/test_parallel.py
    '';

there are more probems in py3k + darwin tests btw..